### ISSUE TITLE
[Feature Request] CISO support and Disc Read Emulation thereof

### ISSUE THREAD

**Issue by kingjinxy** [emukidid/swiss-gc#468]
Hello,

While Swiss supports NKIT (good compression, but inaccurate disc timings), ISO (no compression, but fully accurate), and scrubbed (irreversible compression, not recommended), it does not seem to support the GameCube .ciso format.

As far as I know, CISO is identical to ISO, but according to this page (https://wiki.gbatemp.net/wiki/Nintendont_Compatibility_List; the section "Shrunk"), it removes the empty blocks that do not contain game data but preserves the file offsets. It also apparently has no effect on disc seek timing. I do not know for sure, but could this mean that Swiss users could have the best of NKIT and ISO?

That is to say that CISO has the smaller files associated with NKIT (though slightly larger, in my experience) but none of the disadvantages, like inaccurate disc seek timing.

Thank you for your consideration.

**Comment by Extrems** [comment]
Yes, this is already planned and I had pushed for CISO in NKit v2.

**Comment by kingjinxy** [comment]
Other than (https://gbatemp.net/threads/nkit-recover-to-redump-images-shrink-and-preserve-wii-and-gamecube-disc-images-as-playable.548479/page-9), is there some place where I can watch the development of NKit, or is it more of a “it’ll release when it’s done” project?

Also, what exactly does CISO support in NKit imply? I thought that they were two similar but ultimately different formats.

**Comment by Extrems** [comment]
Development is happening in private in the game preservation community.

**Comment by 60fpshacksrock** [comment]
@Extrems
Another, lighter format is zso lz4 if cso is too heavy so to speak. It has been implemented in Open PS2 Loader and works very well. 

See here: https://www.psx-place.com/threads/testers-needed-open-ps2-loader-with-zso-support-compressed-iso.37223/

**Comment by Extrems** [comment]
I've already been aware of it for 2 weeks. It's interesting but it's still going to be difficult to squeeze into our limited memory footprint.

**Comment by Extrems** [comment]
I also need to clarify since you brought this up here, that CSO and CISO are not the same thing. CISO is not actually compressed.

**Comment by 60fpshacksrock** [comment]
I noticed that very recently about CISO with Wii Backup Manager. It's basically just trimmed similar to wbfs right?

**Comment by Extrems** [comment]
They're basically the same thing, but WBFS has weird baggage from originally being, uh, a file system.

**Comment by 60fpshacksrock** [comment]
Yeah I used it back then when I used the neogamma loader.

Thankfully FAT32 took priority over it. 

NKit is great too. 
I haven't read too much into it, but I figured since it allows Wii games to be compressed much better that it decrypts the game data.

**Comment by databonanza** [comment]
I just tried @ToyCollectorCR -- they do not work within the most recent version of Swiss for GameCube.
